### ISSUE TITLE
IO Error when reading Parquet file that works fine on Pandas.

### ISSUE THREAD

**Issue by luizberti** [duckdb/duckdb#7083]
### What happens?

I'm transcoding some public data dumps from StackExchange from XML to Parquet using Rust's `parquet` crate (`v37.0.0`) from the official `arrow-rs` repository. Some of my tables work fine with DuckDB but others don't and the reason for that is not very clear, especially since all of them work (or seem to work) on Pandas.

The error is the following, and on all Parquet's where this happens it's an off-by-one as is this one:
```error
Error: IO Error: DELTA_BYTE_ARRAY - length mismatch between values and byte array lengths
(attempted read of 14337 from 14336 entries) - corrupt file?
```

### To Reproduce

1. `CREATE VIEW users AS SELECT * FROM read_parquet('users.parquet');`
2. `SUMMARIZE SELECT * FROM users;`

If you give me some way of sending the file over to you I can definitely do that, given that this is all information from public data dumps anyways. It's a 550MB download (the smallest of my tables I can repro this on).

I've also attached my code that does the transcoding, it's in the ZIP file below and `cargo build --release` on the latest Rust stable should build it just fine. In the ZIP there are also some sample files for which the transcoding works fine and DuckDB will parse the output Parquet without issue.

To transcode something, run the following (if there's a parsing error try adding `--bom` as well):
```bash
cargo run -- convert xml --schema=tags.dumparq --output=tags.parquet --input=Tags.xml --pluck=row
```

[transcoding.zip](https://github.com/duckdb/duckdb/files/11225626/transcoding.zip)

### OS:

Linux amd64

### DuckDB Version:

Latest Stable but also validated against the master nightly build

### DuckDB Client:

CLI

### Full Name:

Luiz Berti

### Affiliation:

Unaffiliated

### Have you tried this on the latest `master` branch?

- [X] I agree

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] I agree

**Comment by ryanrupp** [comment]
I was just about to log something for this :) - the error message comes from https://github.com/duckdb/duckdb/pull/6412 - I'm able to reproduce with a string column (binary) that spans multiple pages with `DELTA_BYTE_ARRAY` encoding.

Given the query
```
select * from '<the_parquet_file>'
```

[10k_rows_single_page.parquet](https://github.com/ryanrupp/duckdb/blob/parquet-delta-byte-array-encoding-issue/10k_rows_single_page.parquet) ==> works
[100k_rows_multiple_pages.parquet](https://github.com/ryanrupp/duckdb/blob/parquet-delta-byte-array-encoding-issue/100k_rows_multiple_pages.parquet) ==> `Error: IO Error: DELTA_BYTE_ARRAY - length mismatch between values and byte array lengths (attempted read of 16301 from 16300 entries) - corrupt file?`

Using `parquet` CLI

```
parquet pages 100k_rows_multiple_pages.parquet 

Column: row_number
--------------------------------------------------------------------------------
  page   type  enc  count   avg size   size       rows     nulls   min / max
  0-0    data  _ _  100000  8.00 B     781.254 kB 100000   0       "1.0" / "100000.0"


Column: json_column
--------------------------------------------------------------------------------
  page   type  enc  count   avg size   size       rows     nulls   min / max
  0-0    data  _ D  16300   38.53 B    613.261 kB 16300    0       
  0-1    data  _ D  16300   38.53 B    613.276 kB 16300    0       
  0-2    data  _ D  16300   38.53 B    613.267 kB 16300    0       
  0-3    data  _ D  16300   38.53 B    613.339 kB 16300    0       
  0-4    data  _ D  16300   38.52 B    613.187 kB 16300    0       
  0-5    data  _ D  16300   38.53 B    613.369 kB 16300    0       
  0-6    data  _ D  2200    38.55 B    82.830 kB  2200     0       
```

vs the single page version that works

```
Column: row_number
--------------------------------------------------------------------------------
  page   type  enc  count   avg size   size       rows     nulls   min / max
  0-0    data  _ _  10000   8.00 B     78.129 kB  10000    0       "1.0" / "10000.0"


Column: json_column
--------------------------------------------------------------------------------
  page   type  enc  count   avg size   size       rows     nulls   min / max
  0-0    data  _ D  10000   38.53 B    376.225 kB 10000    0    
```

Also you can avoid hitting this by keeping under that page number to an extent (maybe before fetching the next page triggers?) i.e.

```
-- works
select * from '100k_rows_multiple_pages.parquet' limit 14336;

-- fails with: Error: IO Error: DELTA_BYTE_ARRAY - length mismatch between values and byte array lengths (attempted read of 17139 from 17138 entries) - corrupt file?
select * from '100k_rows_multiple_pages.parquet' limit 14337;
```

These files were generated `parquet-mr` `1.12.3` and can be read by various other query engines.

**Comment by Mytherin** [comment]
Thanks for the report and the reproducible examples! I have pushed a fix in #7112. Indeed the issue was related to the multiple delta byte array pages - the `delta_offset` was not correctly reset when a new page was read.
